### ISSUE TITLE
Investigate array pooling when submitting mesh/texture geometry to the runtime environment

### ISSUE THREAD

**Issue by Frooxius** [Neos-Metaverse/NeosPublic#159]
Could improve memory usage and reduce GC pressure.

**Comment by Frooxius** [comment]
It seems Unity is updating API's to support their NativeArray, but it's not complete yet. Will wait a bit to see where they go with that.

**Comment by shiftyscales** [comment]
Since the creation of this issue, work has been done on the mesh API.

Details of what has been done to the mesh API in 2019.3 can be found in this Google Doc: https://docs.google.com/document/d/1I225X6jAxWN0cheDz_3gnhje3hWNMxTZq3FZQs5KqPc

Additionally, they have proposed improvements to be implemented into Unity 2020.1:
https://docs.google.com/document/d/1QC7NV7JQcvibeelORJvsaTReTyszllOlxdfEsaVL2oA

**Comment by Frooxius** [comment]
Yes, I plan on integrating the new Mesh API. I've done some preliminary work and testing, but it needs more time to be integrated.

**Comment by shiftyscales** [comment]
Sounds good Froox. I was mostly commenting as a means of sharing what I found on the topic while educating myself on the specifics of what this issue was about!
